firefox-ios/Client/Frontend/Browser/MainMenu/Views/MainMenuDetailsViewController.swift

```diff
@@ -121,12 +110,87 @@ class MainMenuDetailViewController: UIViewController,
         submenuContent.detailHeaderView.adjustLayout()
     }
 
-    private func setupTableView() {
-        reloadTableView(with: submenuData)
+    private func setupCallbacks() {
+        submenuContent.detailHeaderView.backToMainMenuCallback = { [weak self] in
+            guard let self else { return }
+            store.dispatch(
+                MainMenuAction(
+                    windowUUID: self.windowUUID,
+                    actionType: MainMenuDetailsActionType.backToMainMenu
+                )
+            )
+        }
+        submenuContent.detailHeaderView.dismissMenuCallback = { [weak self] in
+            guard let self else { return }
+            store.dispatch(
+                MainMenuAction(
+                    windowUUID: self.windowUUID,
+                    actionType: MainMenuDetailsActionType.dismissView
+                )
+            )
+        }
+    }
+
+    private func refreshContent() {
+        submenuContent.setViews(with: submenuState.title, and: .KeyboardShortcuts.Back)
+        reloadTableView(with: submenuState.menuElements)
+    }
+
+    // MARK: - Redux
+    func subscribeToRedux() {
+        store.dispatch(
+            ScreenAction(
+                windowUUID: windowUUID,
+                actionType: ScreenActionType.showScreen,
+                screen: .mainMenuDetails
+            )
+        )
+
+        let uuid = windowUUID
+        store.subscribe(self, transform: {
+            return $0.select({ appState in
+                return MainMenuDetailsState(appState: appState, uuid: uuid)
+            })
+        })
+    }
+
+    func unsubscribeFromRedux() {
+        store.dispatch(
+            ScreenAction(
+                windowUUID: windowUUID,
+                actionType: ScreenActionType.closeScreen,
+                screen: .mainMenuDetails
+            )
+        )
+    }
+
+    func newState(state: MainMenuDetailsState) {
+        submenuState = state
+
+        refreshContent()
```

Comment:
Do we want to call this method only on a certain state update? It seems this reloads the entire table view, so just wanted to make sure we only want to reload when needed.

Reply:
I see what you're saying. I think it's small enough that it kinda doesn't matter - and as far as I understand, we shouldn't be optimizing views updating themselves with the same information. I'll ask in the channel, just to get clarification, and then I'll update accordingly.

Also, I meant to have this at the end of the function, so if you're taking an available action, you're not actually refreshing one last time for no reason. I'll make that change.